src/core/statistics/useStatistics.ts

```diff
@@ -0,0 +1,60 @@
+// Useful resources for Mapp tracking:
+// https://documentation.mapp.com/1.0/en/manual-track-request-25105181.html
+// https://documentation.mapp.com/1.0/en/how-to-send-manual-tracking-requests-page-updates-7240681.html
+
+export type CustomClickParameter = {
+  [key: number]:
+    | number
+    | { [key: number]: string }
+    | { [key: string]: string[] };
+};
+
+export interface EventData {
+  [key: string]: string | number | Record<string, unknown>;
+}
+export interface EventDataWithCustomClickParameter extends EventData {
+  customClickParameter: CustomClickParameter;
+}
+
+// 1. page - when a page request is simulated
+// 2. click - for measuring actions that don't cause page load;
+// 3. link - clicking a link that triggers a new page load
+// 4. pageupdate - information on the page changes without a new page load
+export type EventType = "page" | "click" | "link" | "pageupdate";
+
+export type TrackParameters = {
+  id: number;
+  name: string;
+  trackedData: string | number | string[];
+};
+
+export function useStatistics() {
+  // If the global wts object doesn't exist, it means we are in dev environment.
+  // Here instead of actually tracking we just log the data to the console.
+  if (!window.wts) {
+    window.wts = {
+      push(trackingProps: ["send", EventType, EventData]) {
```

Comment:
Why not also creating a type for the first parameter, like the second and third?
Could for instance be called: `Action`

Reply:
Good point, why not! :) We now have a type for it 😉